src/server/ability.ts

```diff
@@ -1,42 +1,42 @@
 import {AbilityBuilder, Ability} from '@casl/ability'
-import {loadCurrentUser} from 'lib/users'
 import {intersection, isString} from 'lodash'
+import {loadCurrentViewerRoles} from '../lib/viewer'
 
 type Actions = 'manage' | 'upload'
 type Subjects = 'Video' | 'all'
+export type Roles = 'admin' | 'editor' | 'publisher'
 type AppAbility = Ability<[Actions, Subjects]>
 
 export async function getAbilityFromToken(token: string) {
-  const user = await loadCurrentUser(token)
-  return defineAbilityFor(user)
+  const viewerRoles = await loadCurrentViewerRoles(token)
+  return defineAbilityFor(viewerRoles)
 }
 
-function defineAbilityFor(user: any) {
-  // AbilityBuilder https://casl.js.org/v5/en/guide/define-rules#ability-builder-class
+/**
+ * @see {@link https://casl.js.org/v5/en/guide/define-rules#ability-builder-class|AbilityBuilder}
```

Comment:
jsdoc is pretty sweet

you can actually replace TS with it which is interesting